src/java_bytecode/java_object_factory.cpp

```diff
@@ -1019,9 +1030,12 @@ void gen_nondet_init(
   //   <type> <identifier>;
   for(const symbolt * const symbol_ptr : symbols_created)
   {
-    code_declt decl(symbol_ptr->symbol_expr());
-    decl.add_source_location()=loc;
-    init_code.add(decl);
+    if(!symbol_ptr->is_static_lifetime)
+    {
+      code_declt decl(symbol_ptr->symbol_expr());
```

Comment:
Nit:Pull this into a function to avoid duplication with line [969](https://github.com/diffblue/cbmc/pull/1156/commits/65835c249097f2be8905a66df5e117902f418406#diff-233653d9ec20537d004eef631225bfffR969)